src/org/skypro/skyshop/basket/ProductBasket.java

```diff
@@ -14,60 +14,41 @@ public void add(Product product) {
     }
 
     public List<Product> removeByName(String name) {
-        List<Product> removedProducts = new ArrayList<>();
-        List<Product> matchingProducts = products.getOrDefault(name, Collections.emptyList());
-        removedProducts.addAll(matchingProducts);
-        products.remove(name);
-        return removedProducts;
+        List<Product> removeProducts = products.remove(name);
+        if (removeProducts == null) {
+            return new ArrayList<>();
```

Comment:
лучше вернуть пустой список, а не массив из 10 элементов

```suggestion
            return Collections.emptyList();
```